### PR TITLE
Allow delegating cancellation requests to eligible flow services

### PR DESCRIPTION
**Pull request by andreas-borglin**

### CODE REVIEW

**Review by iapetus98 — APPROVED** [review]

File: PaymentServiceSample/src/main/java/com/aevi/sdk/pos/flow/paymentservicesample/service/PaymentService.java [hunk 1]
```diff
@@ -86,6 +92,20 @@ protected void subscribeToFlowServiceEvents(BaseStageModel model, Class<? extend
         });
     }
 
+    private void handleCancelOrResume(BaseStageModel model, Class<? extends Activity> activityToRestart) {
+        // For illustration, cancel immediately for TRANSACTION_PROCESSING but resume for others
+        if (model instanceof TransactionProcessingModel) {
```
Comment (brettwold) [inline PaymentServiceSample/src/main/java/com/aevi/sdk/pos/flow/paymentservicesample/service/PaymentService.java:97]: Should we provide a nice easy way to get whether or not it was a cancel or resume?

File: flow-base-api/src/main/java/com/aevi/sdk/flow/model/config/FlowApp.java [hunk 2]
```diff
@@ -30,6 +31,7 @@
     private final String id;
     private final boolean mandatory;
     private final String conditionalOn;
+    private boolean delegateCancellationsTo;
```
Comment (brettwold) [inline flow-base-api/src/main/java/com/aevi/sdk/flow/model/config/FlowApp.java:34]: final?

**Review by brettwold — APPROVED** [review]
Ok just a question about if we provide something friendly for flow services to know it was a cancel or resume

File: PaymentServiceSample/src/main/java/com/aevi/sdk/pos/flow/paymentservicesample/service/PaymentService.java (hunk 1 above)
Comment (andreas-borglin) [reply PaymentServiceSample/src/main/java/com/aevi/sdk/pos/flow/paymentservicesample/service/PaymentService.java:97]: Hmm...what do you mean? `cancelOrResume` is a new event, `resume` remains as before

File: flow-base-api/src/main/java/com/aevi/sdk/flow/model/config/FlowApp.java (hunk 2 above)
Comment (andreas-borglin) [reply flow-base-api/src/main/java/com/aevi/sdk/flow/model/config/FlowApp.java]: fixed

File: PaymentServiceSample/src/main/java/com/aevi/sdk/pos/flow/paymentservicesample/service/PaymentService.java (hunk 1 above)
Comment (andreas-borglin) [reply PaymentServiceSample/src/main/java/com/aevi/sdk/pos/flow/paymentservicesample/service/PaymentService.java:97]: updated

Comment (brettwold) [reply PaymentServiceSample/src/main/java/com/aevi/sdk/pos/flow/paymentservicesample/service/PaymentService.java:97]: Lovely

**Review by brettwold — APPROVED** [review]

**Pull request merged**